crates/autopilot/src/domain/settlement/solution/mod.rs

```diff
@@ -0,0 +1,280 @@
+//! This module defines Solution as originated from a mined transaction
+//! calldata.
+
+use {
+    self::trade::Trade,
+    crate::domain::{
+        auction::{self},
+        competition,
+        eth,
+        fee,
+        OrderUid,
+    },
+    std::collections::HashMap,
+};
+
+mod tokenized;
+mod trade;
+pub use error::Error;
+
+/// A solution that was executed on-chain.
+///
+/// Contains only data observable on-chain. No off-chain data is used to create
+/// this struct.
+///
+/// Referenced as [`settlement::Solution`] in the codebase.
+#[derive(Debug)]
+pub struct Solution {
+    trades: Vec<Trade>,
+    /// Data that was appended to the regular call data of the `settle()` call
+    /// as a form of on-chain meta data. This is used to associate a
+    /// solution with an auction for which this solution was picked as a winner.
+    auction_id: auction::Id,
```

Comment:
Given that we can't really do anything with this struct without having access to the actual auction (all public functions require some aspect of the auction, either prices or orders), I still think we should come up with a plan of how to construct it with an actual auction instance as a field (instead of just the ids). But not super important and definitely not something for this PR.

Reply:
If we want this, I would prefer to:
1. make `Transaction::input` field of a special `settlement::Calldata` instead of general `eth::Calldata`.
2. have a method `settlement::Calldata::get_auction() -> Option<AuctionId>`
3. Do this as a first thing in the `settlement::Solution` constructor. Then, having an auction id, fetch `Competition` from provided `infra::Persistence` instance.